code/game/objects/items/weapons/clown_items.dm

```diff
@@ -121,12 +121,13 @@
 
 /obj/item/weapon/bikehorn/attack_self(mob/user)
 	if(!spam_flag)
-		spam_flag = 1
+		spam_flag = TRUE
 		playsound(src.loc, honksound, 50, 1)
 		src.add_fingerprint(user)
-		spawn(cooldowntime)
-			spam_flag = 0
-	return
+		addtimer(CALLBACK(src, .proc/clear_cooldown), cooldowntime)
+
+/obj/item/weapon/bikehorn/proc/clear_cooldown()
+	spam_flag = FALSE

```

Comment:
you're not comparing the full costs here.
you're not just checking a bool (because byond doesn't have bools so you're checking a floating point number >= 1), you're creating a callback datum to set to the timing subsystem to look after it for a couple of ticks that then calls a proc.

as opposed to comparing a number when you call a proc.

both are still floating point number comparisons, it's just one has to annoy a very important subsystem to do so, I know which I'd pick in this case.